### PR TITLE
Create new request

### PR DESCRIPTION
**Pull request by oddsson**
Added a screen for creating a request. There is a lot happening here but mainly I'm

- Creating a case when you've entered a police case number and suspect national id
- Created a auto saving mechanism where we save a field in the case when you move the focus out of a input box. F.x. save the suspect name when you move out of the suspect name input field.
- Log unhandled errors to Sentry

### CODE REVIEW

File: apps/judicial-system/web/src/api/index.ts [hunk 1]
```diff
@@ -1,30 +1,86 @@
...
+      return cases
+    } else {
+      throw new Error(response.statusText)
     }
-  } else {
+  } catch (ex) {
     // TODO: Log error
+    console.log(ex)
+  }
+}
```
Comment (gudjong) [inline apps/judicial-system/web/src/api/index.ts:36]: Nice error handling

File: apps/judicial-system/web/src/api/index.ts [hunk 2]
```diff
@@ -1,30 +1,86 @@
...
+  caseField: string,
+  caseFieldValue: string,
+) => Promise<number> = async (
+  caseId: string,
+  caseField: string,
+  caseFieldValue: string,
+) => {
+  if (caseId !== '') {
+    const propertyChange = JSON.parse(`{"${caseField}": "${caseFieldValue}"}`)
+    const response = await fetch(`/api/case/${caseId}`, {
```
Comment (gudjong) [inline apps/judicial-system/web/src/api/index.ts:73]: Will you never be saving multiple fields at the same time?
For example, when previous saves failed?

File: apps/judicial-system/web/src/routes/CreateDetentionRequest/CreateDetentionRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      if (response === 200) {
+        // Assign new value to the field the user is changing
+        copyOfWorkingCase.case[caseField] = caseFieldValue
+
+        // Update the working case
+        setWorkingCase(copyOfWorkingCase)
+      } else {
+        setAutoSaveSucceded(false)
+
+        // TODO: Do something when autosave fails
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/CreateDetentionRequest/CreateDetentionRequest.tsx:78]: Exactly, perhaps collect fields to save?

File: package.json [hunk 4]
```diff
@@ -61,6 +61,8 @@
     "@opentelemetry/tracing": "0.8.3",
     "@sentry/browser": "5.17.0",
     "@sentry/node": "5.17.0",
+    "@sentry/react": "^5.22.3",
+    "@sentry/tracing": "^5.22.3",
```
Comment (gudjong) [inline package.json:65]: Interesting, are you the first to use sentry in a react app?

File: apps/judicial-system/web/src/api/index.ts (hunk 1 above)
Comment (oddsson) [reply apps/judicial-system/web/src/api/index.ts:36]: Hehh 😅  My thinking is that this is all going to change when we start using GraphQL so I didn't make this my nr.1 priority.

File: package.json (hunk 4 above)
Comment (oddsson) [reply package.json:65]: Apparently 🤷‍♂️  But Sentry is the official way to go according to 👉 https://github.com/island-is/handbook/blob/master/docs/adr/0005-error-tracking-and-monitoring.md

File: apps/judicial-system/web/src/api/index.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/web/src/api/index.ts:36]: Hmm, not sure what I was commenting on here, but remember distinctly that I liked that you were catching all error :)

But we will start with REST, not GraphQL, so at some time we need to do something with the error

**Review by gudjong — APPROVED** [review]
I think this is ready for merge.

File: apps/judicial-system/web/src/api/index.ts (hunk 2 above)
Comment (oddsson) [reply apps/judicial-system/web/src/api/index.ts:73]: It depends on if we are going to opt for the solution to display an error message at the top of the screen, saying that something went wrong, try again. In that solution we'd collect all the fields that were not saved and save them together. If we opt for the "this specific input did not save approach" with a checkmark icon for inputs that are saved and an error icon for those that didn't, we might want to save just that field. Did we decide to implement the first solution when we talked about this earlyer?

Comment (gudjong) [reply apps/judicial-system/web/src/api/index.ts:73]: No, we did not decide on anything.

**Pull request merged**